### PR TITLE
checker: prevent the regions in split-cache from becoming the target of merge

### PR DESCRIPTION
**Pull request by HunDunDM**
### What is changed and how it works?

* checker: prevent the regions in split-cache from becoming the target of merge

### Check List

Tests

- Unit test

Related changes

- Need to cherry-pick to the release branch

### Release note

- checker: prevent the regions in split-cache from becoming the target of merge

### CODE REVIEW

**Comment by HunDunDM** [comment]
Originally, a unittest tested this bug, but due to many problems in the test code, the unittest failed to detect the bug.

File: server/schedule/checker/merge_checker_test.go [hunk 1]
```diff
@@ -122,8 +122,8 @@ func (s *testMergeCheckerSuite) SetUpTest(c *C) {
 				},
 			},
 			&metapb.Peer{Id: 109, StoreId: 4},
-			core.SetApproximateSize(10),
-			core.SetApproximateKeys(10),
+			core.SetApproximateSize(1),
```
Comment (HunDunDM) [inline server/schedule/checker/merge_checker_test.go:125]: Here, the original value 10 is greater than threshold 2, which causes the region to be unable to be merged.

File: server/schedule/checker/merge_checker_test.go [hunk 2]
```diff
@@ -191,10 +191,19 @@ func (s *testMergeCheckerSuite) TestBasic(c *C) {
 	c.Assert(ops, NotNil)
 	c.Assert(ops[0].RegionID(), Equals, s.regions[2].GetID())
 	c.Assert(ops[1].RegionID(), Equals, s.regions[1].GetID())
-	s.cluster.RuleManager.DeleteRule("test", "test")
+	s.cluster.RuleManager.DeleteRule("pd", "test")
```
Comment (HunDunDM) [inline server/schedule/checker/merge_checker_test.go:194]: The rule was not deleted correctly here, which caused the subsequent test to be invalid.

File: server/schedule/checker/merge_checker_test.go [hunk 3]
```diff
@@ -191,10 +191,19 @@ func (s *testMergeCheckerSuite) TestBasic(c *C) {
...
 
 	// Skip recently split regions.
 	s.cluster.SetSplitMergeInterval(time.Hour)
+	ops = s.mc.Check(s.regions[2])
+	c.Assert(ops, IsNil)
+
+	s.mc.startTime = time.Now().Add(-2 * time.Hour)
```
Comment (HunDunDM) [inline server/schedule/checker/merge_checker_test.go:201]: The original test did not modify the `startTime`. Due to the following code logic, all regions cannot be merged.

```
	expireTime := m.startTime.Add(m.opts.GetSplitMergeInterval())
	if time.Now().Before(expireTime) {
		checkerCounter.WithLabelValues("merge_checker", "recently-start").Inc()
		return nil
	}
```

**Review by lhy1024 — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]
LGTM

**Comment by hi-rustin** [comment]
> /run-integration-lightning-test

I think maybe this CI broken.

**Comment by hi-rustin** [comment]
> /run-integration-lightning-test

Or only "run-all-tests" can trigger it.

**Pull request merged**